### PR TITLE
Fixed VirtualEnvironmentExists

### LINKED ISSUE #127: Query EnvironmentExists is now broken
#### Description:

The query EnvironmentExists does not work properly (it tries to run a process with the .exe instead of checking if the executable exists)

#### How to replicate:

Run the LBT environment install component.

#### Expected behaviour:

python environment should be installed

### PR DESCRIPTION
**Pull request by Tom-Kingstone**
### Issues addressed by this PR

Closes #127 

Added a check for if the executable it is trying to run actually exists before running.

### Test files

run LBT environment installer in grasshopper.

### CODE REVIEW

**Comment by Tom-Kingstone** [comment]
@BHoMBot check compliance
@BHoMBot check core

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Based on discussion with @jamesramsden-bh and @Tom-Kingstone , this is not the right place to add the protection - it needs adding in the `Version` method as well.

**Comment by Tom-Kingstone** [comment]
@BHoMBot check compliance
@BHoMBot check required

File: Python_Engine/Query/Version.cs [hunk 1]
```diff
@@ -48,12 +48,19 @@ public static PythonVersion Version(string pythonExecutable)
                 }
             };
             string versionString;
-            using (Process p = Process.Start(process.StartInfo))
+            try
```
Comment (FraserGreenroyd) [inline Python_Engine/Query/Version.cs:51]: This method should check if the file exists before even making the `Process` object.

File: Python_Engine/Query/VirtualEnvironment.cs [hunk 2]
```diff
@@ -44,7 +44,9 @@ public static bool VirtualEnvironmentExists(string envName, PythonVersion python
             bool directoryExists = Directory.Exists(VirtualEnvironmentDirectory(envName));
             bool executableExists = File.Exists(VirtualEnvironmentExecutable(envName));
             bool kernelExists = Directory.Exists(VirtualEnvironmentKernel(envName));
-            bool versionMatches = Version(VirtualEnvironmentExecutable(envName)) == pythonVersion;
+            bool versionMatches = false;
+            if (executableExists)
+                versionMatches = Version(VirtualEnvironmentExecutable(envName)) == pythonVersion;
```
Comment (FraserGreenroyd) [inline Python_Engine/Query/VirtualEnvironment.cs:49]: The output from `VirtualEnvironmentExecutable()` should be stored and reused as discussed given it's called twice in this method currently.

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Review by FraserGreenroyd — APPROVED** [review]
That looks robust now on code review @Tom-Kingstone , thanks!

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**